core/domain/skill_domain.py

```diff
@@ -910,6 +910,27 @@ def update_rubrics_from_model(cls, versioned_rubrics, current_version):
 
         versioned_rubrics['rubrics'] = updated_rubrics
 
+    def get_all_image_filenames_during_creation(self):
+        """Returns the filenames for all images in the skill as present during
+        creation i.e from the skill explanation and rubrics.
+
+        Returns:
+            list(str). The list of filenames.
+        """
+        all_rte_components = html_cleaner.get_rte_components(
+            self.skill_contents.explanation.html)
+
+        for rubric in self.rubrics:
+            for explanation in rubric.explanations:
+                all_rte_components.extend(
+                    html_cleaner.get_rte_components(explanation))
+
+        filenames = [
+            component['customization_args']['filepath-with-value']
+            for component in all_rte_components
+            if component and component['id'] == 'oppia-noninteractive-image']
```

Comment:
I'm don't think its good to handle HTML/RTE internals in the domain layer. Can we do this:
1. skill.get_all_subtitled_htmls() [Not sure about the naming]
2. The caller method (here the handler or skill service) can call html_validation_service.get_image_filenames(htmls)

Let me know what you think?

Reply:
Done, did it similar to how we do for explorations using a function in '_services' and '_domain'.